src/nth.js

```diff
@@ -8,14 +8,16 @@
  */
 
 /**
- * Returns `nth` (starts from `0` of course) item of an `iterable`.
+ * Returns `nth` (starts from `0` of course) item of the given `iterable`.
+ * Consumes all items of the generator.
+ * Consumes all items up to the index of the iterator, including the returned item.
```

Comment:
it consumes the generator and it doesn't consume the iterator if the iterator doesn't implement the `return() method`, like e.g. Array Iterator. http://www.ecma-international.org/ecma-262/7.0/#sec-iteratorclose. I can add it to the comments to clarify what's going on.